src/backend/services/metrics.py

```diff
@@ -297,17 +298,54 @@ def wrapper(self, query: str, documents: Dict[str, Any], **kwargs: Any) -> Any:
     return wrapper
 
 
+def collect_compass_metrics(func_name: str, method: str):
+    def report_metrics(func: Callable) -> Callable:
+        @wraps(func)
+        def wrapper(self, parameters: Dict[str, Any], **kwargs: Any) -> Any:
+            start_time = time.perf_counter()
+            metrics_data, kwargs = _initialize_sdk_metrics_data(
+                func_name, method, None, **kwargs
+            )
+
+            response_dict = {}
+            try:
+                response = func(self, parameters, **kwargs)
+                response_dict = to_dict(response)
+            except Exception as e:
+                metrics_data = handle_error(metrics_data, e)
+                raise e
+            finally:
+                metrics_data.duration_ms = time.perf_counter() - start_time
+                run_loop(metrics_data)
+                return response_dict
+
+        return wrapper
+
+    return report_metrics
+
+
 def initialize_sdk_metrics_data(
```

Comment:
Can you unify these two functions? You can change the existing decorators to also receive the name and method!